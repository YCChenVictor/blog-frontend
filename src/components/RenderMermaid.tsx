import React, { useRef, useState, useEffect } from "react";
import mermaid from "mermaid";

const RenderMermaid = (props: React.PropsWithChildren) => {
  const markId = useRef(`mark${Math.floor(Math.random() * 100000 + 1)}`);
  const [svg, setSvg] = useState("");
  useEffect(() => {
    const renderMermaid = async () => {
      const svg = await mermaid.render(markId.current, String(props.children));
      setSvg(svg.svg);
    };
    renderMermaid().catch((error) => {
      console.log(error);
    });
  }, []);
  return <div dangerouslySetInnerHTML={{ __html: svg }}></div>;
};

export default RenderMermaid;
