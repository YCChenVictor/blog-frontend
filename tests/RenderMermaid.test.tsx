import * as renderer from 'react-test-renderer';
import * as React from 'react';
import RenderMermaid from '../src/components/RenderMermaid';

describe('RenderMermaid', () => {
  it('renders Article component without crashing', () => {
    const component = renderer.create(
      <RenderMermaid />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
