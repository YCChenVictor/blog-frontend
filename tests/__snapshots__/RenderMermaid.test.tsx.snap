// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMermaid renders Article component without crashing 1`] = `
<div
  dangerouslySetInnerHTML={
    {
      "__html": "",
    }
  }
/>
`;
